conans/model/editable_cpp_info.py

```diff
@@ -1,73 +1,73 @@
 
+    @staticmethod
+    def load(filepath):
         parser = configparser.ConfigParser(allow_no_value=True)
         parser.optionxform = str
-        parser.read(filepath)
+        try:
+            parser.read(filepath)
+        except configparser.Error:
+            raise ConanException("Error parsing layout file: %s" % filepath)
+        data = OrderedDict()
+        for section in parser.sections():
+            pkg, key = section.split(":", 1) if ':' in section else (None, section)
+            if key not in EditableCppInfo.cpp_info_dirs:
+                raise ConanException("Wrong cpp_info field '%s' in layout file: %s"
+                                     % (key, filepath))
+            data.setdefault(pkg, {})[key] = [k for k, _ in parser.items(section)]
 
-        if not require_namespace:
-            ret = {k: [] for k in cls.cpp_info_dirs}
-            for section in ret.keys():
-                if section in parser.sections():
-                    ret[section] = [k for k, v in parser.items(section)]  # keys used as values
-        else:
-            ret = defaultdict(lambda: {k: [] for k in cls.cpp_info_dirs})
-            for section in parser.sections():
-                if ':' in section:
-                    namespace, key = section.split(':', 1)
-                    if key in cls.cpp_info_dirs:
-                        # keys used as values
-                        ret[namespace][key] = [k for k, v in parser.items(section)]
-        return EditableCppInfo(ret, uses_namespace=require_namespace)
+        return EditableCppInfo(data)
 
     @staticmethod
-    def _work_on_item(value, base_path, settings, options):
-        value = value.replace('\\', '/')
-        isabs = ntpath.isabs(value) or posixpath.isabs(value)
-        if base_path and not isabs:
-            value = os.path.abspath(os.path.join(base_path, value))
-        value = os.path.normpath(value)
+    def _work_on_item(value, settings, options):
         value = value.format(settings=settings, options=options)
+        value = value.replace('\\', '/')
         return value
 
-    def has_info_for(self, pck_name, use_wildcard=True):
-        if self._uses_namespace:
-            return pck_name in self._data or (use_wildcard and self.WILDCARD in self._data)
-        else:
-            return True
-
-    def apply_to(self, pkg_name, cpp_info, base_path, settings=None, options=None, use_wildcard=True):
-        if self._uses_namespace:
-            if pkg_name in self._data:
-                data_to_apply = self._data[pkg_name]
-            elif use_wildcard and self.WILDCARD in self._data:
-                data_to_apply = self._data[self.WILDCARD]
-            else:
-                data_to_apply = {k: [] for k in self.cpp_info_dirs}
-        else:
-            data_to_apply = self._data
+    def apply_to(self, ref, cpp_info, settings=None, options=None):
+        d = self._data
+        data = d.get(str(ref)) or d.get(None) or {}
```

Comment:
this is called using `ref=node.ref`, does this `str(ref)` include the revision?

Reply:
No, the str (repr) doesn't use the revision at all.